### PR TITLE
fix array indexing bug

### PR DESCRIPTION
**Pull request by EpigeneMax**
## Description and Context
See [CR-391].

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] No code (non-breaking change that does not impact code: formatting, build system, documentation...)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Release (version number increment, possibly with documentation updates)

[CR-391]: https://epigenelabsteam.atlassian.net/browse/CR-391?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/deseq2/test_outlier.py [hunk 1]
```diff
@@ -105,3 +106,21 @@ def test_outlier_filtering_small_counts(self):
         res = dds.results()
         self.assertFalse(np.isnan(res["pvalue"].iloc[0]))
         self.assertTrue(np.all(np.isnan(res.pvalue[1:2])))
+
+    def test_CRXXX(self):
+        """TODO"""
```
Comment (soleneweill) [inline tests/deseq2/test_outlier.py:111]: missing test explanation

File: tests/deseq2/test_outlier.py [hunk 2]
```diff
@@ -105,3 +106,21 @@ def test_outlier_filtering_small_counts(self):
...
+                    [33, 439, 9, 2114, 199, 115, 47, 0, 425, 19],
+                    [34, 652, 20, 3040, 308, 153, 62, 2, 543, 34],
+                    [42, 752, 14, 3796, 349, 215, 67, 1, 658, 30],
+                    [38, 565, 15, 2743, 266, 118, 58, 5, 512, 28],
+                ]
+            ),
+            clinicalData=pd.DataFrame({"group": [0, 0, 1, 1, 1, 1]}),
+            design="~ group",
+        )
+        DESeq(dds)
```
Comment (soleneweill) [inline tests/deseq2/test_outlier.py:126]: The test is not testing anything, does it ?

**Review by soleneweill — CHANGES_REQUESTED** [review]

File: tests/deseq2/test_outlier.py (hunk 1 above)
Comment (EpigeneMax) [reply tests/deseq2/test_outlier.py]: indeed...
fixed :)

File: tests/deseq2/test_outlier.py (hunk 2 above)
Comment (EpigeneMax) [reply tests/deseq2/test_outlier.py:129]: The test checks that no exception is raised during computation, but it does not check the results against a reference -- that's the job of other tests

Comment (soleneweill) [reply tests/deseq2/test_outlier.py:129]: ok

**Review by soleneweill — APPROVED** [review]

**Pull request merged**